src/provider_asn1_util.erl

```diff
@@ -71,3 +73,6 @@ set_arg(State, Key, Val) ->
     Args = rebar_state:get(State, asn1_args, []),
     ArgsMap = maps:from_list(Args),
     rebar_state:set(State, asn1_args, maps:to_list(maps:put(Key, Val, ArgsMap))).
+
+asn_basename(ASNFileName) ->
+    filename:basename(filename:basename(filename:basename(ASNFileName, ".asn1"), ".asn"), ".set").
```

Comment:
What is the `".set"` extension here?

Reply:
from https://www.erlang.org/doc/man/asn1ct.html#compile-2

> If it is needed to compile a set of ASN.1 modules into an Erlang file with encode/decode functions, ensure to list all involved files in a configuration file. This configuration file must have a double extension ".set.asn" (".asn" can alternatively be ".asn1" or ".py"). List the input file names within quotation marks (""), one at each row in the file. If the input files are File1.asn, File2.asn, and File3.asn, the configuration file must look as follows:

```
File1.asn
File2.asn
File3.asn
```

But might be better to read the files from the content of those files. I will make an update to it.

Reply:
It's alright, I missed this part:
> The output files in this case get their names from the configuration file. If the configuration file is named SetOfFiles.set.asn, the names of the output files are SetOfFiles.hrl, SetOfFiles.erl, and SetOfFiles.asn1db.

meaning that stripping `.set.asn` part would be right, because it compiles to the filename of the set.